### ISSUE TITLE
Make `safeParse` always return a `data` property

### ISSUE THREAD

**Issue by dcodesdev** [colinhacks/zod#3142]
## Make `safeParse` always return a `data` property

Why? Because everytime I use `safeParse` I have to do this:

```ts
const handle = (data: any) => {
  // Can't destructure `data` here 😔
  const parsed = mySchema.safeParse(data)

  if (!parsed.success) {
    return
  }

  // Have to destructure again here 😔
  const { email, password } = parsed.data
}
```

I would like to do this instead:

```ts
const handle = (payload: any) => {
  // Can destructure everything here 😎
  const { success, data, error } = mySchema.safeParse(payload)

  if (!success) {
    /**
     * In this case:
     * - `data` is `null`
     * - `error` is a `ZodError` instance
     * - `success` is `false`
     */
    return
  }

  /**
   * In this case:
   * - `data` is the parsed value
   * - `error` is `null`
   * - `success` is `true`
   */
  const { email, password } = data
}
```

This way, all the properties on the return value of `safeParse` can be destructured and used directly instead of having to do the `if` check and then destructuring the `data` property.

**Comment by JacobWeisenburger** [comment]
perhaps this would be helpful:
https://github.com/JacobWeisenburger/zod_utilz#spr

**Comment by joaoGabriel55** [comment]
I can work on that @JacobWeisenburger.

**Comment by joaoGabriel55** [comment]
> perhaps this would be helpful: https://github.com/JacobWeisenburger/zod_utilz#spr
Could we add this zod lib to the repo, to avoid code duplication?

**Comment by colinhacks** [comment]
Fixed in Zod 3.23
